tuxemon/states/pc_locker/__init__.py

```diff
@@ -441,21 +444,21 @@ def find_monster_box(itm: Item, box: list) -> Optional[Item]:
                 for ele in box:
                     if ele.slug == itm.slug:
                         return ele
+                return None
 
             if box:
-                if find_monster_box(itm, box):
-                    if diff <= 0:
-                        stored = player.find_item_in_storage(
-                            find_monster_box(itm, box).instance_id
-                        )
-                        stored.quantity += quantity
-                        player.remove_item(itm)
-                    else:
-                        stored = player.find_item_in_storage(
-                            find_monster_box(itm, box).instance_id
-                        )
-                        stored.quantity += quantity
-                        itm.quantity = diff
+                retrieve = find_monster_box(itm, box)
+                if retrieve is not None:
+                    assert retrieve
```

Comment:
Not going to point this out any more. Please check all your asserts.

Reply:
The if check covers this, so the assert is pointless

Reply:
you're right, I went in goblin mode with asserts, I'm going to fix it